### PR TITLE
add library_annotations; remove package_prefixed_library_names

### PR DESCRIPTION
**Pull request by devoncarew**
- add `library_annotations` (closes https://github.com/dart-lang/lints/issues/177)
- remove `package_prefixed_library_names` (closes https://github.com/dart-lang/lints/issues/172)

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by parlough — APPROVED** [review]

**Review by goderbauer — APPROVED** [review]
LGTM

File: rules.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <!-- core -->
 | Lint Rules | Description | [Fix][] |
 | :--------- | :---------- | ------- |
-| [`avoid_empty_else`](https://dart.dev/lints/avoid_empty_else) | Avoid empty else statements. | ✅ |
+| [`avoid_empty_else`](https://dart.dev/lints/avoid_empty_else) | Avoid empty statements in else clauses. | ✅ |
```
Comment (lrhn) [inline rules.md:8]: in -> as

(Does the lint only target `;`, or also `{}`?)

File: tool/rules.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
   },
   {
     "name": "avoid_empty_else",
-    "description": "Avoid empty else statements.",
+    "description": "Avoid empty statements in else clauses.",
```
Comment (lrhn) [inline tool/rules.json:14]: in -> as

File: rules.md (hunk 1 above)
Comment (devoncarew) [reply rules.md:8]: I believe the ultimate system-of-record for this info are the lint descriptions at https://github.com/dart-lang/sdk/tree/main/pkg/linter/lib/src/rules. Any changes there will flow here (in a few steps).

**Pull request merged**